test/libsolidity/syntaxTests/multiSource/multiple_imports_multiple_alais.sol

```diff
@@ -0,0 +1,17 @@
+==== Source: s1.sol ====
+function f() pure returns (uint) { return 1337; }
+function g() pure returns (uint) { return 42; }
+==== Source: s2.sol ====
+import {f as g} from "s1.sol";
+==== Source: s3.sol ====
+// imports f()->1337 as g()
+import "s2.sol";
+// imports f()->1337 as f() and
+// g()->42 as g
+import {f as f, g as g} from "s1.sol";
```

Comment:
This should issue a shadowing warning since alias `g` is shadowed by `f`'s namesake alias

Reply:
The error message can be made more precise

Reply:
Error message improvement is tracked here now: https://github.com/ethereum/solidity/issues/9878